packages/core/contracts/proxy-scripts/lsp-broker/LspUniswapV2Broker.sol

```diff
@@ -15,27 +18,197 @@ import "../../financial-templates/long-short-pair/LongShortPair.sol";
+    // of tokens, considering the mint and trade,is equal to the pool ratio after the trade. This is computed using the
+    // following logic: Assume m tokens are minted in equal proportion of long and short. This contract will therefore
+    // hold t_l=m+△l long tokens and t_s=m-△s after a trade of s for l tokens. The ratio t_l/t_s must equal the pool
+    // ratio after the trade. The pool ratio, considering the trade can be expressed as (R_a-△a)(R_b+△bλ)=k for a
+    // trade of △a for △b and a swap fee of λ=(1-swapFee). Using this, with a bit of algebra and simplification, we can
+    // solve for the resultant ratio to be (m+△a)/(m-△b)=(R_a-△a)/(R_b+△b). i.e the ratio of mint+trade must equal the
+    // ratio of token A and B in the pool after the trade has concluded. We also know that △a can be solved for as
+    // △a=(△bλR_a)/(R_b+△bλ) by manipulating a known uniswap equation. Using this equation and the expression of ratios
+    // we can solve simaltaniously for △b. I.e how many short tokens do we need to sell such that the ratio of long and
+    // short after the trade equal the pool ratios. Numerically, this works out to this form:
+    // △b=(sqrt(R_b(R_a+m)(R_a(R_b(λ^2+2λ+1)+4λm)+R_b(λ^2-2λ+1)m))+R_b(-λ-1)(R_a+m))/(2λ(R_a+m)). For how this was solved
+    // from the algebraic solution above see https://bit.ly/3jn7AF6 on wolfram alpha showing the derivation.
+    function computeSwapToMintAtPoolRatio(
+        FixedPoint.Signed memory m,
+        FixedPoint.Signed memory ra,
+        FixedPoint.Signed memory rb
+    ) private pure returns (bool, uint256) {
+        FixedPoint.Signed memory numerator1 =
+            sqrt(
+                rb.mul(ra.add(m)).mul(
+                    ra
+                        .mul(rb.mul(lambda2().add(num(2).mul(lambda())).add(num(1))).add(num(4).mul(lambda().mul(m))))
+                        .add(rb.mul(lambda2().sub(num(2).mul(lambda())).add(num(1))).mul(m))
+                )
+            );
+
+        FixedPoint.Signed memory numerator2 = rb.mul(ra.add(m)).mul(num(-1).sub(lambda()));
+
+        FixedPoint.Signed memory numerator = numerator1.mul(1e9).add(numerator2);
+
+        FixedPoint.Signed memory denominator = num(2).mul(lambda()).mul(ra.add(m));
+
+        FixedPoint.Signed memory tradeSize = numerator.div(denominator);
+
+        // If the trade size is negative then we are traversing the equation backwards. In this case the contract must
+        // swap between swapTokensForExactTokens and swapExactTokensForTokens methods as we are always solving for △b.
+        // In other words, we are always solving for how many short tokens are needed to be bought/sold. If the number
+        // is positive then we are buying them. if it is negative then we are selling them. In the case we are selling
+        // them (negative) then we swap the polarity and apply the swap fee to the trade.
+        bool aToB = tradeSize.isLessThan(0);
```

Comment:
Its awesome that this function returns positive/negative depending on the direction you need, and the magnitude is always correct!